rest_framework/utils/encoders.py

```diff
@@ -65,3 +65,14 @@ def default(self, obj):
         elif hasattr(obj, '__iter__'):
             return tuple(item for item in obj)
         return super().default(obj)
+
+
+class CustomScalar:
+    """
+    CustomScalar that knows how to encode timedelta that renderer
+    can understand.
+    """
+    @classmethod
+    def represent_timedelta(cls, dumper, data):
+        value = str(data.total_seconds())
+        return dumper.represent_scalar('tag:yaml.org,2002:str', value)
```

Comment:
are you using represent_scalar from existing code?

Reply:
it is actually the function of yaml Dumper class for overriding the tag for different datatypes or python object. 
so this function gives custom representation for timedelta and first argument is dumper obj where we are adding this function